kpi/tests/test_mock_data_exports.py

```diff
@@ -655,3 +665,58 @@ def test_export_with_disabled_questions(self):
         # fails with `KeyError` prior to fix for kobotoolbox/formpack#219
         self.run_csv_export_test(expected_lines, asset=asset)
 
+    def test_anotheruser_can_export_when_submissions_publicly_shared(self):
+        """
+        Running through behaviour described in issue kpi/#2870 where an asset
+        that has been publicly shared and then explicity shared with a user, the
+        user has lower permissions than an anonymous user and is therefore
+        unable to export submission data.
+        """
+        # resetting permissions of `anotheruser` to have no permissions
+        self.asset.remove_perm(self.anotheruser, PERM_PARTIAL_SUBMISSIONS)
+        self.asset.remove_perm(self.anotheruser, PERM_VIEW_ASSET)
+
+        anonymous_user = get_anonymous_user()
+
+        assert self.asset.has_perm(self.anotheruser, PERM_VIEW_ASSET) == False
+        assert PERM_VIEW_ASSET not in self.asset.get_perms(self.anotheruser)
+
+        self.asset.assign_perm(self.anotheruser, PERM_VIEW_ASSET)
+        self.asset.assign_perm(self.anotheruser, PERM_CHANGE_ASSET)
```

Comment:
Why `self.anotheruser` is assigned `PERM_CHANGE_ASSET`? ~~They do not seem to need it anywhere in the test?~~ 
`PERM_CHANGE_ASSET` is needed to export. 
What do you think about moving the line just before the export (with a comment why it is needed)?

Reply:
Sounds good 👍  I set those permissions as they were the ones specified in the issue that Tino applied.

Reply:
I removed the `PERM_VIEW_ASSET` as mentioned in the previous comment about implied permissions, but if I move the assignment of `PERM_CHANGE_ASSET` to just before the export then the tests checking for `PERM_VIEW_ASSET` fail. So I kept it where it was, but just added a test to ensure that `PERM_CHANGE_ASSET` is on the asset so that it's not out of place and added a comment about needing it for exports.

Reply:
Only `view_submissions`  is needed to export data. User don't need `change_asset`. You can remove it and related assertions to this permission.